src/node_options-inl.h

```diff
@@ -39,6 +39,19 @@ void OptionsParser<Options>::AddOption(const std::string& name,
                               help_text});
 }
 
+template <typename Options>
+void OptionsParser<Options>::AddOption(const std::string& name,
+                                       const std::string& help_text,
+                                       uint32_t Options::* field,
+                                       OptionEnvvarSettings env_setting) {
+  options_.emplace(
+      name,
+      OptionInfo{kInteger,
```

Comment:
Yes, this is bogus and will lead to platform-specific behaviour. `kInteger` currently stands for 64-bit integer, including the assumption that the target pointer can be cast to `int64_t*`. For introducing `uint32_t` (or maybe, preferably, `uint64_t`?) support, you’d need a new enum value and handle that everywhere where all enum values are handled individually.

(The reason this fails on linuxone is that it’s a big-endian platform, so the casting between `int64_t*` and  `uint32_t*` will give different behaviour than on little-endian machines.)

Reply:
> Yes, this is bogus and will lead to platform-specific behaviour. `kInteger` currently stands for 64-bit integer, including the assumption that the target pointer can be cast to `int64_t*`. For introducing `uint32_t` (or maybe, preferably, `uint64_t`?) support, you’d need a new enum value and handle that everywhere where all enum values are handled individually.

I think we should handle `uint64_t`, as it's what is used by llhttp. Then, we'll cast it to `unit32_t` when passing it down to `http_parser`. Can you add a couple of pointers on how to add that new enum?

Reply:
> Can you add a couple of pointers on how to add that new enum?

I don’t think it’s much different from grepping around for `kInteger` and adding analogue code paths with `uint64_t` instead of `int64_t` in the places where it is used